core/src/main/java/utilcalc/core/reportGen/ReportGenUtil.java

```diff
@@ -0,0 +1,59 @@
+package utilcalc.core.reportGen;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import utilcalc.core.model.input.ServiceCost;
+
+final class ReportGenUtil {
+
+    private ReportGenUtil() {}
+
+    static List<ServiceCost> splitServiceCostsByReport(
+            LocalDate reportStartDate, LocalDate reportEndDate, List<ServiceCost> serviceCosts) {
+        List<ServiceCost> result = new ArrayList<>();
+
+        for (ServiceCost cost : serviceCosts) {
+            LocalDate costStartDate = cost.startDate();
+            LocalDate costEndDate = cost.endDate();
+
+            LocalDate overlapStart =
+                    reportStartDate.isAfter(costStartDate) ? reportStartDate : costStartDate;
+            LocalDate overlapEnd =
+                    reportEndDate.isBefore(costEndDate) ? reportEndDate : costEndDate;
+
+            if (overlapStart.isBefore(overlapEnd)) {
```

Comment:
Podle code coverage je tahle podminka vzdycky true, mozna nam chybi nejaky test case, ktery by to pokryl? (neoveroval jsem to)